### PR TITLE
Fixes #36438 - configure API request limit to avoid memory leaks

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: spec/classes/pulpcore_spec.rb [hunk 1]
```diff
@@ -162,6 +164,21 @@
...
+          {
+            api_service_worker_max_requests: 100,
+            api_service_worker_max_requests_jitter: 10
+          }
+        end
+
+        it do
+          is_expected.to compile.with_all_deps
+          is_expected.to contain_systemd__unit_file('pulpcore-api.service').with_content(%r{--max-requests 100})
+          is_expected.to contain_systemd__unit_file('pulpcore-api.service').with_content(%r{--max-requests-jitter 10})
```
Comment (ekohl) [inline spec/classes/pulpcore_spec.rb:178]: You can chain them:
```suggestion
          is_expected.to contain_systemd__unit_file('pulpcore-api.service').with_content(%r{--max-requests 100}).with_content(%r{--max-requests-jitter 10})
```

File: spec/classes/pulpcore_spec.rb [hunk 2]
```diff
@@ -150,6 +150,8 @@
           is_expected.to contain_pulpcore__socket_service('pulpcore-api')
           is_expected.to contain_systemd__unit_file('pulpcore-api.socket')
           is_expected.to contain_systemd__unit_file('pulpcore-api.service').with_content(%r{--workers 2})
+          is_expected.to contain_systemd__unit_file('pulpcore-api.service').with_content(%r{--max-requests 50})
+          is_expected.to contain_systemd__unit_file('pulpcore-api.service').with_content(%r{--max-requests-jitter 30})
```
Comment (ekohl) [inline spec/classes/pulpcore_spec.rb:154]: You can chain them:
```suggestion
          is_expected.to contain_systemd__unit_file('pulpcore-api.service').with_content(%r{--max-requests 50}).with_content(%r{--max-requests-jitter 30})
```

**Review by ekohl — APPROVED** [review]

File: manifests/init.pp [hunk 3]
```diff
@@ -233,6 +239,8 @@
   Integer[0] $api_service_worker_count = min(4, $facts['processors']['count']) + 1,
   Integer[0] $content_service_worker_timeout = 90,
   Integer[0] $api_service_worker_timeout = 90,
+  Integer[0] $api_service_worker_max_requests = 50,
```
Comment (dralley) [inline manifests/init.pp:242]: @evgeni We discovered the "real" root cause now, so we can relax this number significantly.  I would back it off to a much higher value, like say 1000 for the limit and 200 for the jitter.

Comment (ehelms) [reply manifests/init.pp:242]: Do you have a link to info about the real root cause?

Comment (dralley) [reply manifests/init.pp:242]: See the discussion near the end on the BZ https://bugzilla.redhat.com/show_bug.cgi?id=2122872

**Pull request merged**